third_party/move/move-vm/runtime/src/frame.rs

```diff
@@ -180,11 +222,22 @@ impl Frame {
     }
 
     pub(crate) fn local_ty_at(&self, idx: usize) -> &Type {
-        &self.local_tys[idx]
+        match &self.local_tys {
+            LocalTys::None => {
+                unreachable!("Local types are not used when there are no runtime type checks")
```

Comment:
Can you locally reason that this panic never happens?

Reply:
Before we had `vec![]` and index, so behaviour is the same, but yes, locally `None` is created only when checks are off. Let me add a unit test for that then 👍